<!DOCTYPE html><html lang="en" ><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="generator" content="Jekyll v4.2.2" /><meta property="og:title" content="Generate Frequency count table and motif logo from given fasta sequences" /><meta name="author" content="Dewan Shrestha" /><meta property="og:locale" content="en" /><meta name="description" content="To generate a motif logo first we need to create a count table for the nucleotide bases from fasta sequences and then information matrix to plot motif logo" /><meta property="og:description" content="To generate a motif logo first we need to create a count table for the nucleotide bases from fasta sequences and then information matrix to plot motif logo" /><link rel="canonical" href="/posts/generate-motif-logo/" /><meta property="og:url" content="/posts/generate-motif-logo/" /><meta property="og:site_name" content="Dewan Shrestha" /><meta property="og:type" content="article" /><meta property="article:published_time" content="2022-09-27T18:30:00+00:00" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Generate Frequency count table and motif logo from given fasta sequences" /><meta name="twitter:site" content="@dewshrs" /><meta name="twitter:creator" content="@Dewan Shrestha" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Dewan Shrestha"},"dateModified":"2022-09-27T18:30:00+00:00","datePublished":"2022-09-27T18:30:00+00:00","description":"To generate a motif logo first we need to create a count table for the nucleotide bases from fasta sequences and then information matrix to plot motif logo","headline":"Generate Frequency count table and motif logo from given fasta sequences","mainEntityOfPage":{"@type":"WebPage","@id":"/posts/generate-motif-logo/"},"url":"/posts/generate-motif-logo/"}</script><title>Generate Frequency count table and motif logo from given fasta sequences | Dewan Shrestha</title><link rel="apple-touch-icon" sizes="180x180" href="/assets/img/favicons/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/assets/img/favicons/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/assets/img/favicons/favicon-16x16.png"><link rel="manifest" href="/assets/img/favicons/site.webmanifest"><link rel="shortcut icon" href="/assets/img/favicons/favicon.ico"><meta name="apple-mobile-web-app-title" content="Dewan Shrestha"><meta name="application-name" content="Dewan Shrestha"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/assets/img/favicons/browserconfig.xml"><meta name="theme-color" content="#ffffff"><link rel="preconnect" href="https://fonts.googleapis.com" ><link rel="dns-prefetch" href="https://fonts.googleapis.com" ><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="dns-prefetch" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://fonts.googleapis.com" ><link rel="dns-prefetch" href="https://fonts.googleapis.com" ><link rel="preconnect" href="https://cdn.jsdelivr.net" ><link rel="dns-prefetch" href="https://cdn.jsdelivr.net" ><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Lato&family=Source+Sans+Pro:wght@400;600;700;900&display=swap"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4/dist/css/bootstrap.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.11.2/css/all.min.css"><link rel="stylesheet" href="/assets/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@1.0.1/dist/bootstrap-toc.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/magnific-popup@1/dist/magnific-popup.min.css"> <script src="https://cdn.jsdelivr.net/npm/jquery@3/dist/jquery.min.js"></script> <script type="text/javascript"> class ModeToggle { static get MODE_KEY() { return "mode"; } static get MODE_ATTR() { return "data-mode"; } static get DARK_MODE() { return "dark"; } static get LIGHT_MODE() { return "light"; } static get ID() { return "mode-toggle"; } constructor() { if (this.hasMode) { if (this.isDarkMode) { if (!this.isSysDarkPrefer) { this.setDark(); } } else { if (this.isSysDarkPrefer) { this.setLight(); } } } let self = this; /* always follow the system prefers */ this.sysDarkPrefers.addEventListener("change", () => { if (self.hasMode) { if (self.isDarkMode) { if (!self.isSysDarkPrefer) { self.setDark(); } } else { if (self.isSysDarkPrefer) { self.setLight(); } } self.clearMode(); } self.notify(); }); } /* constructor() */ get sysDarkPrefers() { return window.matchMedia("(prefers-color-scheme: dark)"); } get isSysDarkPrefer() { return this.sysDarkPrefers.matches; } get isDarkMode() { return this.mode === ModeToggle.DARK_MODE; } get isLightMode() { return this.mode === ModeToggle.LIGHT_MODE; } get hasMode() { return this.mode != null; } get mode() { return sessionStorage.getItem(ModeToggle.MODE_KEY); } /* get the current mode on screen */ get modeStatus() { if (this.isDarkMode || (!this.hasMode && this.isSysDarkPrefer)) { return ModeToggle.DARK_MODE; } else { return ModeToggle.LIGHT_MODE; } } setDark() { $('html').attr(ModeToggle.MODE_ATTR, ModeToggle.DARK_MODE); sessionStorage.setItem(ModeToggle.MODE_KEY, ModeToggle.DARK_MODE); } setLight() { $('html').attr(ModeToggle.MODE_ATTR, ModeToggle.LIGHT_MODE); sessionStorage.setItem(ModeToggle.MODE_KEY, ModeToggle.LIGHT_MODE); } clearMode() { $('html').removeAttr(ModeToggle.MODE_ATTR); sessionStorage.removeItem(ModeToggle.MODE_KEY); } /* Notify another plugins that the theme mode has changed */ notify() { window.postMessage({ direction: ModeToggle.ID, message: this.modeStatus }, "*"); } } /* ModeToggle */ const toggle = new ModeToggle(); function flipMode() { if (toggle.hasMode) { if (toggle.isSysDarkPrefer) { if (toggle.isLightMode) { toggle.clearMode(); } else { toggle.setLight(); } } else { if (toggle.isDarkMode) { toggle.clearMode(); } else { toggle.setDark(); } } } else { if (toggle.isSysDarkPrefer) { toggle.setLight(); } else { toggle.setDark(); } } toggle.notify(); } /* flipMode() */ </script><body data-spy="scroll" data-target="#toc" data-topbar-visible="true"><div id="sidebar" class="d-flex flex-column align-items-end"><div class="profile-wrapper text-center"><div id="avatar"> <a href="/" class="mx-auto"> <img src="https://pbs.twimg.com/profile_images/1106327178382397440/yCl9Agfr_400x400.jpg" alt="avatar" onerror="this.style.display='none'"> </a></div><div class="site-title mt-3"> <a href="/">Dewan Shrestha</a></div><div class="site-subtitle font-italic">Bioinformatics</div></div><ul class="w-100"><li class="nav-item"> <a href="/" class="nav-link"> <i class="fa-fw fas fa-home ml-xl-3 mr-xl-3 unloaded"></i> <span>HOME</span> </a><li class="nav-item"> <a href="/categories/" class="nav-link"> <i class="fa-fw fas fa-stream ml-xl-3 mr-xl-3 unloaded"></i> <span>CATEGORIES</span> </a><li class="nav-item"> <a href="/tags/" class="nav-link"> <i class="fa-fw fas fa-tag ml-xl-3 mr-xl-3 unloaded"></i> <span>TAGS</span> </a><li class="nav-item"> <a href="/archives/" class="nav-link"> <i class="fa-fw fas fa-archive ml-xl-3 mr-xl-3 unloaded"></i> <span>ARCHIVES</span> </a><li class="nav-item"> <a href="/about/" class="nav-link"> <i class="fa-fw fas fa-info-circle ml-xl-3 mr-xl-3 unloaded"></i> <span>ABOUT</span> </a></ul><div class="sidebar-bottom mt-auto d-flex flex-wrap justify-content-center align-items-center"> <button class="mode-toggle btn" aria-label="Switch Mode"> <i class="fas fa-adjust"></i> </button> <span class="icon-border"></span> <a href="https://github.com/dewshr" aria-label="github" target="_blank" rel="noopener"> <i class="fab fa-github"></i> </a> <a href="https://twitter.com/dewshrs" aria-label="twitter" target="_blank" rel="noopener"> <i class="fab fa-twitter"></i> </a> <a href=" javascript:location.href = 'mailto:' + ['dewshrs','gmail.com'].join('@')" aria-label="email" > <i class="fas fa-envelope"></i> </a> <a href="/feed.xml" aria-label="rss" > <i class="fas fa-rss"></i> </a></div></div><div id="topbar-wrapper"><div id="topbar" class="container d-flex align-items-center justify-content-between h-100 pl-3 pr-3 pl-md-4 pr-md-4"> <span id="breadcrumb"> <span> <a href="/"> Home </a> </span> <span>Generate Frequency count table and motif logo from given fasta sequences</span> </span> <i id="sidebar-trigger" class="fas fa-bars fa-fw"></i><div id="topbar-title"> Post</div><i id="search-trigger" class="fas fa-search fa-fw"></i> <span id="search-wrapper" class="align-items-center"> <i class="fas fa-search fa-fw"></i> <input class="form-control" id="search-input" type="search" aria-label="search" autocomplete="off" placeholder="Search..."> </span> <span id="search-cancel" >Cancel</span></div></div><div id="main-wrapper" class="d-flex justify-content-center"><div id="main" class="container pl-xl-4 pr-xl-4"><div class="row"><div id="core-wrapper" class="col-12 col-lg-11 col-xl-9 pr-xl-4"><div class="post pl-1 pr-1 pl-md-2 pr-md-2"><h1 data-toc-skip>Generate Frequency count table and motif logo from given fasta sequences</h1><div class="post-meta text-muted"> <span> Posted <em class="" data-ts="1664303400" data-df="ll" data-toggle="tooltip" data-placement="bottom"> Sep 27, 2022 </em> </span><div class="d-flex justify-content-between"> <span> By <em> Dewan Shrestha </em> </span><div> <span class="readtime" data-toggle="tooltip" data-placement="bottom" title="221 words"> <em>1 min</em> read</span></div></div></div><div class="post-content"><p>To generate a motif logo first we need to create a count table for the nucleotide bases from fasta sequences and then information matrix to plot motif logo</p><h1 id="create-count-table-from-fasta-sequence">Create count table from fasta sequence</h1><div class="language-py highlighter-rouge"><div class="code-header"> <span data-label-text="Python"><i class="fas fa-code small"></i></span> <button aria-label="copy" data-title-succeed="Copied!"><i class="far fa-clipboard"></i></button></div><div class="highlight"><code><table class="rouge-table"><tbody><tr><td class="rouge-gutter gl"><pre class="lineno">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
</pre><td class="rouge-code"><pre>
<span class="kn">import</span> <span class="nn">logomaker</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="n">pd</span>
<span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">motifs</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="n">sns</span>

<span class="n">seq_list</span> <span class="o">=</span> <span class="p">[]</span>

<span class="n">fasta</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">input_fasta_file</span><span class="p">,</span><span class="s">'r'</span><span class="p">).</span><span class="n">readlines</span><span class="p">()</span>

<span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">fasta</span><span class="p">:</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">line</span><span class="p">.</span><span class="n">startswith</span><span class="p">(</span><span class="s">'&gt;'</span><span class="p">):</span>
			<span class="n">seq_list</span><span class="p">.</span><span class="n">append</span><span class="p">(</span><span class="n">line</span><span class="p">.</span><span class="n">upper</span><span class="p">().</span><span class="n">replace</span><span class="p">(</span><span class="s">'</span><span class="se">\n</span><span class="s">'</span><span class="p">,</span><span class="s">''</span><span class="p">))</span>

	<span class="k">print</span><span class="p">(</span><span class="s">'...... getting base counts .....</span><span class="se">\n</span><span class="s">'</span><span class="p">)</span>

	<span class="n">instances</span> <span class="o">=</span> <span class="p">[</span><span class="n">Seq</span><span class="p">(</span><span class="n">x</span><span class="p">.</span><span class="n">upper</span><span class="p">())</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">seq_list</span><span class="p">]</span>
	<span class="n">m</span> <span class="o">=</span> <span class="n">motifs</span><span class="p">.</span><span class="n">create</span><span class="p">(</span><span class="n">instances</span><span class="p">)</span>

	<span class="n">m_df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">m</span><span class="p">.</span><span class="n">counts</span><span class="p">)</span>
    <span class="n">m_df</span> <span class="o">=</span> <span class="n">m_df</span><span class="p">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">10</span><span class="p">,:]</span>
    <span class="n">m_df</span><span class="p">.</span><span class="n">head</span><span class="p">()</span>

<span class="c1"># Running above for loop gives following output
</span>    <span class="n">A</span>	<span class="n">C</span>	<span class="n">G</span>	<span class="n">T</span>
<span class="mi">0</span>	<span class="mi">239</span>	<span class="mi">239</span>	<span class="mi">351</span>	<span class="mi">593</span>
<span class="mi">1</span>	<span class="mi">513</span>	<span class="mi">231</span>	<span class="mi">211</span>	<span class="mi">467</span>
<span class="mi">2</span>	<span class="mi">281</span>	<span class="mi">140</span>	<span class="mi">771</span>	<span class="mi">230</span>
<span class="mi">3</span>	<span class="mi">284</span>	<span class="mi">451</span>	<span class="mi">443</span>	<span class="mi">244</span>
<span class="mi">4</span>	<span class="mi">410</span>	<span class="mi">621</span>	<span class="mi">156</span>	<span class="mi">235</span>
</pre></table></code></div></div><h1 id="generate-information-matrix-to-generate-motif-logo">Generate information matrix to generate motif logo</h1><div class="language-py highlighter-rouge"><div class="code-header"> <span data-label-text="Python"><i class="fas fa-code small"></i></span> <button aria-label="copy" data-title-succeed="Copied!"><i class="far fa-clipboard"></i></button></div><div class="highlight"><code><table class="rouge-table"><tbody><tr><td class="rouge-gutter gl"><pre class="lineno">1
2
3
4
5
6
7
8
9
10
11
12
13
</pre><td class="rouge-code"><pre><span class="n">t_df</span> <span class="o">=</span> <span class="n">logomaker</span><span class="p">.</span><span class="n">transform_matrix</span><span class="p">(</span><span class="n">m_df</span><span class="p">,</span> <span class="n">from_type</span> <span class="o">=</span> <span class="s">'counts'</span><span class="p">,</span> <span class="n">to_type</span><span class="o">=</span><span class="s">'information'</span><span class="p">)</span>


<span class="n">sns</span><span class="p">.</span><span class="nb">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.5</span><span class="p">,</span> <span class="n">style</span><span class="o">=</span><span class="s">'white'</span><span class="p">)</span>
<span class="n">plt</span><span class="p">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span><span class="mi">15</span><span class="p">))</span>

<span class="n">crp_logo</span><span class="o">=</span><span class="n">logomaker</span><span class="p">.</span><span class="n">Logo</span><span class="p">(</span><span class="n">t_df</span><span class="p">,</span><span class="n">font_name</span><span class="o">=</span><span class="s">'Arial Rounded MT Bold'</span><span class="p">)</span>
<span class="n">crp_logo</span><span class="p">.</span><span class="n">style_spines</span><span class="p">(</span><span class="n">visible</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">crp_logo</span><span class="p">.</span><span class="n">style_spines</span><span class="p">(</span><span class="n">spines</span><span class="o">=</span><span class="p">[</span><span class="s">'left'</span><span class="p">,</span> <span class="s">'bottom'</span><span class="p">],</span> <span class="n">visible</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

<span class="n">plt</span><span class="p">.</span><span class="n">title</span><span class="p">(</span><span class="s">'motif'</span><span class="p">)</span>
<span class="n">plt</span><span class="p">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">plt</span><span class="p">.</span><span class="n">savefig</span><span class="p">(</span><span class="s">'/Users/dshresth/Downloads/test.pdf'</span><span class="p">)</span>
</pre></table></code></div></div><p><img data-src="/assets/img/test.png" alt="motif" data-proofer-ignore></p><p>Please make sure you have required packages for this to work. I also have the script <a href="https://github.com/dewshr/Bioinformatics-tools-and-data/tree/main/create_motif_logo">here</a>.</p><h2 id="reference"><span class="mr-2">Reference</span><a href="#reference" class="anchor text-muted"><i class="fas fa-hashtag"></i></a></h2><ul><li><a href="https://logomaker.readthedocs.io/en/latest/implementation.html#logo-class">logomaker</a></ul></div><div class="post-tail-wrapper text-muted"><div class="post-meta mb-3"> <i class="far fa-folder-open fa-fw mr-1"></i> <a href='/categories/bioinformatics/'>Bioinformatics</a></div><div class="post-tags"> <i class="fa fa-tags fa-fw mr-1"></i> <a href="/tags/motif/" class="post-tag no-text-decoration" >motif</a> <a href="/tags/logo/" class="post-tag no-text-decoration" >logo</a> <a href="/tags/biopython/" class="post-tag no-text-decoration" >biopython</a> <a href="/tags/pwm/" class="post-tag no-text-decoration" >pwm</a> <a href="/tags/fasta/" class="post-tag no-text-decoration" >fasta</a> <a href="/tags/seaborn/" class="post-tag no-text-decoration" >seaborn</a></div><div class="post-tail-bottom d-flex justify-content-between align-items-center mt-3 pt-5 pb-2"><div class="license-wrapper"> This post is licensed under <a href="https://creativecommons.org/licenses/by/4.0/"> CC BY 4.0 </a> by the author.</div><div class="share-wrapper"> <span class="share-label text-muted mr-1">Share</span> <span class="share-icons"> <a href="https://twitter.com/intent/tweet?text=Generate+Frequency+count+table+and+motif+logo+from+given+fasta+sequences+-+Dewan+Shrestha&url=%2Fposts%2Fgenerate-motif-logo%2F" data-toggle="tooltip" data-placement="top" title="Twitter" target="_blank" rel="noopener" aria-label="Twitter"> <i class="fa-fw fab fa-twitter"></i> </a> <a href="https://www.facebook.com/sharer/sharer.php?title=Generate+Frequency+count+table+and+motif+logo+from+given+fasta+sequences+-+Dewan+Shrestha&u=%2Fposts%2Fgenerate-motif-logo%2F" data-toggle="tooltip" data-placement="top" title="Facebook" target="_blank" rel="noopener" aria-label="Facebook"> <i class="fa-fw fab fa-facebook-square"></i> </a> <a href="https://t.me/share/url?url=%2Fposts%2Fgenerate-motif-logo%2F&text=Generate+Frequency+count+table+and+motif+logo+from+given+fasta+sequences+-+Dewan+Shrestha" data-toggle="tooltip" data-placement="top" title="Telegram" target="_blank" rel="noopener" aria-label="Telegram"> <i class="fa-fw fab fa-telegram"></i> </a> <i id="copy-link" class="fa-fw fas fa-link small" data-toggle="tooltip" data-placement="top" title="Copy link" data-title-succeed="Link copied successfully!"> </i> </span></div></div></div></div></div><div id="panel-wrapper" class="col-xl-3 pl-2 text-muted"><div class="access"><div id="access-lastmod" class="post"><div class="panel-heading">Recently Updated</div><ul class="post-content pl-0 pb-1 ml-1 mt-2"><li><a href="/posts/public-data-links/">Download links for public data and tools</a><li><a href="/posts/creating-genome-index-file/">Create Genome Index file to map sequencing data</a><li><a href="/posts/creating-multiple-subplots-using-seaborn-python/">Creating Multiple Subplots in One Figure Using Seaborn Python</a><li><a href="/posts/add-count-numbers-on-box-plot/">Add frequency count for each category on box plot</a><li><a href="/posts/pandas-tricks/">Pandas Tricks</a></ul></div><div id="access-tags"><div class="panel-heading">Trending Tags</div><div class="d-flex flex-wrap mt-3 mb-1 mr-3"> <a class="post-tag" href="/tags/seaborn/">seaborn</a> <a class="post-tag" href="/tags/genetics/">genetics</a> <a class="post-tag" href="/tags/awk/">awk</a> <a class="post-tag" href="/tags/pandas/">pandas</a> <a class="post-tag" href="/tags/bedtools/">bedtools</a> <a class="post-tag" href="/tags/bioinformatics/">bioinformatics</a> <a class="post-tag" href="/tags/data/">data</a> <a class="post-tag" href="/tags/dna/">DNA</a> <a class="post-tag" href="/tags/download/">download</a> <a class="post-tag" href="/tags/matplotlib/">matplotlib</a></div></div></div><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@1.0.1/dist/bootstrap-toc.min.js"></script><div id="toc-wrapper" class="pl-0 pr-4 mb-5"><div class="panel-heading pl-3 pt-2 mb-2">Contents</div><nav id="toc" data-toggle="toc"></nav></div></div></div><div class="row"><div id="tail-wrapper" class="col-12 col-lg-11 col-xl-9 pl-3 pr-3 pr-xl-4"><div id="related-posts" class="mt-5 mb-2 mb-sm-4"><h3 class="pt-2 mt-1 mb-4 ml-1" data-toc-skip>Further Reading</h3><div class="card-deck mb-4"><div class="card"> <a href="/posts/generate-motif-sequence-from-pwm-matrix/"><div class="card-body"> <em class="small" data-ts="1661194800" data-df="ll" > Aug 22, 2022 </em><h3 class="pt-0 mt-1 mb-3" data-toc-skip>Generate Motif Sequence From Pwm Matrix</h3><div class="text-muted small"><p> I am using PWM (Position Weight Matrix) file in meme format downloaded from JASPAR database as the input file. I am using random.choices from python to randomly generate the nucleotide motif sequen...</p></div></div></a></div><div class="card"> <a href="/posts/creating-multiple-subplots-using-seaborn-python/"><div class="card-body"> <em class="small" data-ts="1661193000" data-df="ll" > Aug 22, 2022 </em><h3 class="pt-0 mt-1 mb-3" data-toc-skip>Creating Multiple Subplots in One Figure Using Seaborn Python</h3><div class="text-muted small"><p> If you want to include multiple plots in a single figure, you can do that by creating axes. Here, is the sample code for that. In this section of code I am just loading the example dataset. impor...</p></div></div></a></div><div class="card"> <a href="/posts/add-count-numbers-on-box-plot/"><div class="card-body"> <em class="small" data-ts="1661205600" data-df="ll" > Aug 22, 2022 </em><h3 class="pt-0 mt-1 mb-3" data-toc-skip>Add frequency count for each category on box plot</h3><div class="text-muted small"><p> Here I will show how to add frequency count to the each category in the box plot. For this we will use the searborn provided example dataset. First, lets load the dataset: import searborn as sns ...</p></div></div></a></div></div></div><div class="post-navigation d-flex justify-content-between"> <a href="/posts/nucleotide_nomenclature/" class="btn btn-outline-primary" prompt="Older"><p>Single letter representation of ambiguous nucleotide</p></a> <a href="/posts/useful-websites/" class="btn btn-outline-primary" prompt="Newer"><p>Biological Data Resources</p></a></div></div></div><footer class="row pl-3 pr-3"><div class="col-12 d-flex justify-content-between align-items-center text-muted pl-0 pr-0"><div class="footer-left"><p class="mb-0"> Â© 2023 <a href="https://twitter.com/dewshrs">Dewan Shrestha</a>. <span data-toggle="tooltip" data-placement="top" title="Except where otherwise noted, the blog posts on this site are licensed under the Creative Commons Attribution 4.0 International (CC BY 4.0) License by the author.">Some rights reserved.</span></p></div><div class="footer-right"><p class="mb-0"> Powered by <a href="https://jekyllrb.com" target="_blank" rel="noopener">Jekyll</a> with <a href="https://github.com/cotes2020/jekyll-theme-chirpy" target="_blank" rel="noopener">Chirpy</a> theme.</p></div></div></footer></div><div id="search-result-wrapper" class="d-flex justify-content-center unloaded"><div class="col-12 col-sm-11 post-content"><div id="search-hints"><div id="access-tags"><div class="panel-heading">Trending Tags</div><div class="d-flex flex-wrap mt-3 mb-1 mr-3"> <a class="post-tag" href="/tags/seaborn/">seaborn</a> <a class="post-tag" href="/tags/genetics/">genetics</a> <a class="post-tag" href="/tags/awk/">awk</a> <a class="post-tag" href="/tags/pandas/">pandas</a> <a class="post-tag" href="/tags/bedtools/">bedtools</a> <a class="post-tag" href="/tags/bioinformatics/">bioinformatics</a> <a class="post-tag" href="/tags/data/">data</a> <a class="post-tag" href="/tags/dna/">DNA</a> <a class="post-tag" href="/tags/download/">download</a> <a class="post-tag" href="/tags/matplotlib/">matplotlib</a></div></div></div><div id="search-results" class="d-flex flex-wrap justify-content-center text-muted mt-3"></div></div></div></div><div id="mask"></div><a id="back-to-top" href="#" aria-label="back-to-top" class="btn btn-lg btn-box-shadow" role="button"> <i class="fas fa-angle-up"></i> </a><div id="notification" class="toast" role="alert" aria-live="assertive" aria-atomic="true" data-animation="true" data-autohide="false"><div class="toast-header"> <button type="button" class="ml-2 ml-auto close" data-dismiss="toast" aria-label="Close"> <span aria-hidden="true">&times;</span> </button></div><div class="toast-body text-center pt-0"><p class="pl-2 pr-2 mb-3">A new version of content is available.</p><button type="button" class="btn btn-primary" aria-label="Update"> Update </button></div></div><script src="https://cdn.jsdelivr.net/npm/simple-jekyll-search@1.10.0/dest/simple-jekyll-search.min.js"></script> <script> SimpleJekyllSearch({ searchInput: document.getElementById('search-input'), resultsContainer: document.getElementById('search-results'), json: '/assets/js/data/search.json', searchResultTemplate: '<div class="pl-1 pr-1 pl-sm-2 pr-sm-2 pl-lg-4 pr-lg-4 pl-xl-0 pr-xl-0"> <a href="{url}">{title}</a><div class="post-meta d-flex flex-column flex-sm-row text-muted mt-1 mb-1"> {categories} {tags}</div><p>{snippet}</p></div>', noResultsText: '<p class="mt-5">Oops! No results found.</p>', templateMiddleware: function(prop, value, template) { if (prop === 'categories') { if (value === '') { return `${value}`; } else { return `<div class="mr-sm-4"><i class="far fa-folder fa-fw"></i>${value}</div>`; } } if (prop === 'tags') { if (value === '') { return `${value}`; } else { return `<div><i class="fa fa-tag fa-fw"></i>${value}</div>`; } } } }); </script> <script src="https://cdn.jsdelivr.net/combine/npm/magnific-popup@1/dist/jquery.magnific-popup.min.js,npm/lozad/dist/lozad.min.js,npm/clipboard@2/dist/clipboard.min.js"></script> <script src="https://cdn.jsdelivr.net/combine/npm/dayjs@1/dayjs.min.js,npm/dayjs@1/locale/en.min.js,npm/dayjs@1/plugin/relativeTime.min.js,npm/dayjs@1/plugin/localizedFormat.min.js"></script> <script defer src="/assets/js/dist/post.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4/dist/js/bootstrap.bundle.min.js"></script> <script defer src="/app.js"></script>
